llvm/lib/CAS/OnDiskCommon.cpp

```diff
@@ -109,6 +109,8 @@ cas::ondisk::tryLockFileThreadSafe(int FD, std::chrono::milliseconds Timeout,
       return std::error_code();
     int Error = errno;
     if (Error == EWOULDBLOCK) {
+      if (Timeout.count() == 0)
```

Comment:
Probably a minor detail, but I see `tryLockFile` returns `errc::no_lock_available`, but this returns `errc::operation_would_block`. Maybe we should change `tryLockFile` to would_block as well as a follow up?